lib/telemetry/telemetry.go

```diff
@@ -0,0 +1,81 @@
+package telemetry
+
+import (
+	"time"
+
+	"github.com/armon/go-metrics"
+)
+
+/// TODO(kit) Label stands in for labels
+type Label struct {
+	Key   string
+	Value string
+}
+
+// PLSREVIEW(kit): I decided to directly wrap the API of go-metrics because it informs all of our call sites.
+//  However, this does not need to be set in stone. For example, I renamed Incr to Inc. This is a good opportunity to
+//  adjust the metrics API we use internally. Any suggestions for improvements would be welcome!
+type MetricsClient interface {
+	// A Gauge should retain the last value it is set to
+	SetGauge(key []string, val float32, labels ...Label)
+
+	// Counters accumulate
+	IncrCounter(key []string, val float32, labels ...Label)
+
+	// Convenience fn to capture durations for samples
+	MeasureSince(key []string, start time.Time, labels ...Label)
+
+	// TODO(kit): I have not thought about filters yet. If we don't update them outside of RuntimeCfg
```

Comment:
I think filtering can be the responsibility of the implementation and not something that we need to expose in the interface.